pkg/auth/providers/common/password.go

```diff
@@ -0,0 +1,61 @@
+package common
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	corev1 "github.com/rancher/types/apis/core/v1"
+	"k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const Namespace = "mgmt-secrets"
+
+func CreateOrUpdateSecrets(secrets corev1.SecretInterface, secretInfo string, field string, authType string) error {
+	if secretInfo != "" {
+		name := fmt.Sprintf("%s-%s", authType, field)
+		secret := &v1.Secret{
```

Comment:
looks like building secret object is repeated in multiple places. Think it would make sense to extract it to its own call, in the common place?